static/app/views/issueDetails/streamline/useSuspectFlags.tsx

```diff
@@ -0,0 +1,116 @@
+import moment from 'moment-timezone';
+
+import type {Event} from 'sentry/types/event';
+import type {Organization} from 'sentry/types/organization';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {useApiQuery} from 'sentry/utils/queryClient';
+
+type RawFlag = {
+  action: string;
+  created_at: string;
+  created_by: string;
+  created_by_type: string;
+  flag: string;
+  id: number;
+  tags: Record<string, any>;
+};
+
+export type RawFlagData = {data: RawFlag[]};
+
+export type FlagSeriesDatapoint = {
+  // flag action
+  label: {formatter: () => string};
+  // flag name
+  name: string;
+  // unix timestamp
+  xAxis: number;
+};
+
+export function hydrateFlagData({
+  rawFlagData,
+}: {
+  rawFlagData: RawFlagData | undefined;
+}): FlagSeriesDatapoint[] {
+  if (!rawFlagData) {
+    return [];
+  }
+
+  // transform raw flag data into series data
+  // each data point needs to be type FlagSeriesDatapoint
+  const flagData = rawFlagData.data.map(f => {
+    return {
+      xAxis: Date.parse(f.created_at),
+      label: {formatter: () => f.action},
+      name: `${f.flag}`,
+    };
+  });
+  return flagData;
+}
+
+export default function useSuspectFlags({
+  organization,
+  query,
+  firstSeen,
+  rawFlagData,
+  event,
+}: {
+  event: Event;
+  firstSeen: string;
+  organization: Organization;
+  query: Record<string, any>;
+  rawFlagData: RawFlagData | undefined;
+}) {
+  const hydratedFlagData: FlagSeriesDatapoint[] = hydrateFlagData({rawFlagData});
+
+  // map flag data to arrays of flag names
+  const auditLogFlagNames = hydratedFlagData.map(f => {
+    return f.name;
+  });
+  const evaluatedFlagNames = event.contexts.flags?.values.map(f => {
+    return f.flag;
+  });
+  const intersection = auditLogFlagNames.filter(f => evaluatedFlagNames?.includes(f));
```

Comment:
Unless there is a flag limit, I would opt to use a set to perform an intersection here. The reason is that this is an On^2 loop that runs on data that is sent by the users so its better to be safe here. Otherwise, a memo should be fine too

Reply:
i used the lodash `intersection` instead, does that work better?
cc @ryan953

Reply:
Yeah, that is fine too, looks like they just iterate over the first set and create a hashmap. I prefer to just write out simple code like that or use something like Set [intersections](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/intersection), but lodash is fine too :)

I would prefer to still memo this sort of stuff, because it could potentially get expensive, but thats up to you to judge :)

Reply:
i'll `useMemo` it!